src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -387,13 +387,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $lowercase = true)
                         $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
                     }
 
-                    if ($strict = $arg->getAttribute('strict')) {
```

Comment:
@stof I meant providing a file that works with both 2.x and 3.x. If we didn't trigger the deprecation with the XML format, it means that we assumed something did set the `strict` option to `false` (otherwise you wouldn't need to provide the option at all). However, in YAML that wouldn't be necessary at all.